WalletWasabi.Gui/Controls/WalletExplorer/SendTabViewModel.cs

```diff
@@ -60,5 +72,55 @@ protected override async Task DoAfterBuildTransaction(BuildTransactionResult res
 
 			ResetUi();
 		}
+
+		[ValidateMethod(nameof(ValidatePayjoinEndPoint))]
+		public string PayjoinEndPoint
+		{
+			get => _payjoinEndPoint;
+			set => this.RaiseAndSetIfChanged(ref _payjoinEndPoint, value);
+		}
+
+		private IPayjoinClient GetPayjoinClient()
+		{
+			if (!string.IsNullOrWhiteSpace(PayjoinEndPoint) && Uri.IsWellFormedUriString(PayjoinEndPoint, UriKind.Absolute))
+			{
+				var payjoinEndPointUri = new Uri(PayjoinEndPoint);
+				return new PayjoinClient(payjoinEndPointUri, Global.TorManager.TorSocks5EndPoint);
+			}
+
+			return null;
+		}
+
+		public ErrorDescriptors ValidatePayjoinEndPoint()
+		{
+			if (string.IsNullOrWhiteSpace(PayjoinEndPoint) || Uri.IsWellFormedUriString(PayjoinEndPoint, UriKind.Absolute))
+			{
+				return ErrorDescriptors.Empty;
+			}
+			return new ErrorDescriptors(new ErrorDescriptor(ErrorSeverity.Error, "Invalid url."));
+		}
+
+		protected override void OnAddressPaste(BitcoinUrlBuilder url)
+		{
+			base.OnAddressPaste(url);
+
+			var payjoinEndPoint = url.UnknowParameters.TryGetValue("bpu", out var endPoint) || url.UnknowParameters.TryGetValue("pj", out endPoint)
+				? endPoint
+				: null;
+
+			if (payjoinEndPoint is { } && (Wallet.KeyManager.IsHardwareWallet || Wallet.KeyManager.IsWatchOnly))
```

Comment:
A hww is always a watchonly wallet so, it is redundant.

```suggestion
			if (payjoinEndPoint is { } && Wallet.KeyManager.IsWatchOnly)
```

Reply:
ack